### PR TITLE
Fabric 7.0: remove Shimmer props and styles from DetailsList

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Description of changes

Removing anything related to `Shimmer` from `DetailsList`, `DetailsRow` and `DetailsRowFields` including props that I marked deprecated in the current master and the styles associated with it. In the deprecation messages, I pointed users to use the `ShimmeredDetailsList` which is acting as a decorator component and does not need `DetailsList` to know about shimmer.

@jdhuntington this change will need to be something to highlight in the release notes.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8903)

### CODE REVIEW

**Comment by Vitalius1** [comment]
@KevinTCoughlin can you please skim through and see if I missed something? Thanks!

@jdhuntington don't forget to include this in the release notes, please.

**Comment by Vitalius1** [comment]
@KevinTCoughlin I would appreciate a quick review to ensure I didn't miss anything. Thanks!

**Comment by KevinTCoughlin** [comment]
I'm a fan of this change-set since it reduces the work our row rendering does. I anticipate some friction for devs when migrating given our usage internally. Let's make sure to clearly document the process, perhaps in our wiki and over-communicate. Rational being perf and API service being acceptable, I think.

Also seems like the toggle state on the following example may be incorrect in `master`,  http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/8903/merge/fabric-website-resources/dist/index.html#/examples/detailslist/Shimmer.

👍 thanks for the follow-thru here and sorry for the delay.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.styles.ts [hunk 1]
```diff
@@ -237,24 +220,7 @@ export const getStyles = (props: IDetailsRowStyleProps): IDetailsRowStyles => {
...
-      },
-
-      // Masking the running shimmer background with borders when it's an Icon placeholder
-      [`&$shimmerIconPlaceholder`]: {
-        borderRight: `${cellStyleProps.cellRightPadding}px solid ${colors.defaultBackground}`,
-        borderBottom: `${(values.compactRowHeight - values.rowShimmerIconPlaceholderHeight) / 2}px solid ${colors.defaultBackground}`,
-        borderTop: `${(values.compactRowHeight - values.rowShimmerIconPlaceholderHeight) / 2}px solid ${colors.defaultBackground}`
-      }
-    }
+    paddingLeft: `${cellStyleProps.cellLeftPadding}px`
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.styles.ts]: We prob can use the raw value here to avoid interpolation

**Pull request merged**